Sources/DatadogSDKTesting/Utils/SwiftExtensions.swift

```diff
@@ -73,26 +73,51 @@ extension Data {
         // #endif
         return result
     }
-
-    var hexString: String {
-        return self.map { String(format: "%02x", $0) }.joined()
+    
+    init?(hex: String) {
```

Comment:
probably worth adding dedicated tests for these two initializers

Reply:
yeah. good idea. added tests and refactored zlib method